### PR TITLE
Changes to support Mattermost 5.9.

### PR DESCRIPTION
**Pull request by 9numbernine9**

### CODE REVIEW

**Comment by drnic** [comment]
I've booted up this PR and can reproduce the failure. The logs aren't immediately helpful:

```
# cat /var/vcap/sys/log/mattermost/mattermost.stderr.log
Error: LoadConfig: utils.config.load_config.decoding.panic,
Usage:
  mattermost [flags]
  mattermost [command]
```

Investigating.

File: jobs/mattermost/templates/config/initial-config.json.erb [hunk 1]
```diff
@@ -134,7 +132,6 @@
         "AtRestEncryptKey": "<%= p("mattermost.SqlSettings.AtRestEncryptKey") %>",
         "QueryTimeout": <%= p("mattermost.SqlSettings.QueryTimeout") %>,
         "ConnMaxLifetimeMilliseconds": <%= p("mattermost.SqlSettings.ConnMaxLifetimeMilliseconds") %>,
-        "EnablePublicChannelsMaterialization": <%= p("mattermost.SqlSettings.EnablePublicChannelsMaterialization") %>
```
Comment (drnic) [inline jobs/mattermost/templates/config/initial-config.json.erb]: In removing this line, you need to remove the comma from the tail of the preceding line. Hurray for JSON strictness :)

Comment (9numbernine9) [reply jobs/mattermost/templates/config/initial-config.json.erb]: _Siiiiiiigh_ do you know how long I started at my commit and looked for _exactly this kind of error_ and didn't find any? :-| I'll fix this in my next commit!

File: jobs/mattermost/spec [hunk 2]
```diff
@@ -540,10 +541,10 @@ properties:
     default: ""
   mattermost.EmailSettings.SendPushNotifications:
     description: Your Mattermost server sends mobile push notifications to the server specified in PushNotificationServer.
-    default: false
+    default: true
```
[the remark this answers is not in the document]
Comment (9numbernine9) [reply jobs/mattermost/spec]: This is an excellent question, and I'm glad you raised it - I find the Mattermost documentation to be a little lacking on this one. Their documentation seems to imply that `false` is the default value here, but the provided `config/default.json` in their source tree has the value set to `true` - and the documentation says that `push-test` is only for Team Edition users; paying customers can use the (ostensibly better) `push.mattermost.com` service instead.

In short - I'm perfectly happy to put this back to `false` because I certainly can't _guarantee_ that this configuration will work in the simplest case, and I also don't want to be known as "The Guy Who Broke Mattermost-Bosh". :-)

**Comment by 9numbernine9** [comment]
Changes made in 7b87b2d1f0a - let's see what happens when `testflight-pr` kicks off again. :-)
